design_docs/crf.md

```diff
@@ -0,0 +1,60 @@
+# Some technical selection in implementation of CRF layer
+## About CRF loss function
+currently the crf loss function is desinged as a seperated method/Class. 
+### Solution 1: standalone loss
+In usage it look like below
+
+```python
+from tensorflow_addons.layers import CRF
+from tensorflow_addons.losses import crf_loss
+
+model = Sequential()
+model.add(Embedding(3001, 300, mask_zero=True)
+
+crf = CRF(10)
+model.add(crf)
+
+model.compile('adam', loss=crf_loss)
+
+model.fit(x, y)
+```
+
+#### pros ####
+the standard way to use loss
+
+#### cons ####
+in the eager mode, there need a complicated patch to make this solution works. 
```

Comment:
Can you provide a minimal example for this?

Reply:
Sure, this is the way currently I implement the CRF layer (code already included in this PR).  I will add more information about how it works.

Reply:
I have updated the documentation to explain where the trick is.